src/ray/raylet/node_manager.cc

```diff
@@ -1707,7 +1707,9 @@ void NodeManager::HandleReturnWorker(const rpc::ReturnWorkerRequest &request,
         HandleDirectCallTaskUnblocked(worker);
       }
       cluster_task_manager_->ReleaseWorkerResources(worker);
-      HandleWorkerAvailable(worker);
+      if (!request.worker_exiting()) {
+        HandleWorkerAvailable(worker);
```

Comment:
Btw, I have a question regarding semantics between `disconnect_worker ` and `worker_exiting `.

Is `disconnect_worker` for forceful termination (since there's something unrecoverable) initiated from the caller whereas `worker_exiting` is for graceful shutdown, and the clean up will be performed by the actor itself (callee), is this correct? 

If this is the right difference, can you comment it from the docstring of `ReturnWorker` method?